.github/workflows/pythonpackage.yml

```diff
@@ -20,21 +20,25 @@ jobs:
           python-version: ${{ matrix.python-version }}
           architecture: x64
       - name: Install and configure Poetry
-        uses: snok/install-poetry@v1.1.1
+        uses: snok/install-poetry@v1.1.6
         with:
-          version: 1.1.4
-          virtualenvs-create: false
+          version: 1.1.6
+          virtualenvs-in-project: true
       - name: Install dependencies
         run: poetry install
         if: steps.cached-poetry-dependencies.outputs.cache-hit != 'true'
       - name: Format check with black
-        run: make format-check
+        run: |
+          source $VENV
+          make format-check
```

Comment:
Ran into some problems with the old setup, so switched from installing dependencies directly inside the runner environment, to installing dependencies to a venv and activating the venv in consequent steps.

Just for context, instead of `source $VENV`, we *could* have used `source .venv/bin/activate`, but since there's a windows combination in the matrix you would need to introduce conditional steps; on Windows you'd want ~`source .venv/scripts/activate`. 

The `$VENV` environment variable is set inside the `install-poetry` action so upstream workflows don't need to worry about which OS you're on 🙂